share_extension/ios/extension_channels.js

```diff
@@ -128,13 +128,22 @@ export default class ExtensionChannels extends PureComponent {
                 return (channel.type === General.DM_CHANNEL && usersInDms.includes(teammateId)) || channel.type === General.GM_CHANNEL;
             });
 
-            const dmProfiles = await Client4.getProfilesByIds(usersInDms);
+            let dmProfiles;
+            if (usersInDms.length) {
+                dmProfiles = await Client4.getProfilesByIds(usersInDms);
+            }
 
             for (let i = 0; i < dms.length; i++) {
                 const channel = dms[i];
                 if (channel.type === General.DM_CHANNEL) {
                     const teammateId = getUserIdFromChannelName(currentUserId, channel.name);
-                    const profile = dmProfiles.find((p) => p.id === teammateId);
+                    let profile;
+                    if (dmProfiles) {
+                        profile = dmProfiles.find((p) => p.id === teammateId);
+                    } else {
+                        const response = await Client4.getProfilesByIds([teammateId]);
```

Comment:
How often does this happen that `dmProfiles` is empty? Would it be a significant improvement if we batched the requests to get the missing teammates? I'm thinking we could look through the DM channels once to get the IDs to load, load them, then go back into this loop

Reply:
I'll see what I can do here